File: official/docs/php/orders/BuyOrder.php
<?php

\EasyPost\EasyPost::setApiKey('EASYPOST_API_KEY');

$order = \EasyPost\Order::retrieve('order_...');

$order->buy([
    'carrier' => 'FedEx',
    'service' => 'FEDEX_GROUND'
]);

echo $order;

File: official/docs/csharp/orders/buy-order.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Newtonsoft.Json;
using EasyPost;

namespace EasyPostExamples
{
    public class Examples
    {
        public static async Task Main()
        {
            string apiKey = Environment.GetEnvironmentVariable("EASYPOST_API_KEY")!;

            EasyPost.ClientManager.SetCurrent(apiKey);

            Order order = await Order.Retrieve("order_...");

            await order.Buy("FedEx", "FEDEX_GROUND");

            Console.WriteLine(JsonConvert.SerializeObject(order, Formatting.Indented));
        }
    }
}
